src/pages/[platform]/build-a-backend/troubleshooting/library-not-configured/index.mdx

```diff
@@ -0,0 +1,149 @@
+* Next.js Page Router: **pages/_app.tsx** or **src/pages/_app.tsx**
+
+<Callout>
+
+If you are using the Next.js App Router, you can follow the suggestions in our [Next.js documentation](/[platform]/build-a-backend/server-side-rendering/nextjs/#configure-amplify-library-for-client-side-usage) for root-level configuration. Keep in mind that if you are calling any APIs at the module-level (i.e. at the top of your file) in any of the Child components, you may still run into this issue.
+
+</Callout>
+
+## Check for module-level Amplify API invocations
+
+When Amplify APIs are used outside of your application lifecycle, there is a risk that a javascript bundler may place that API call before `Amplify.configure`. Module-level function calls (calls at the top-level of a file), are generally evaluated in the order that they are imported.
+
+Below is an example of code that will likely result in a missing configuration or `NoCredentials` error message:
+
+```javascript title="index.ts"
+import { Amplify } from 'aws-amplify';
+import ComponentX from 'module-fetch-auth';
+
+// fetchAuthSession() in ComponentX executed on import
+
+Amplify.configure();
+
+export default function App() {
+  return (
+    <div>
+        <ComponentX />
+    </div>
+  );
+}
+```
+
+```javascript title="module-fetch-auth.ts"
+import { fetchAuthSession } from 'aws-amplify/auth';
+
+fetchAuthSession(); // Will throw "AuthUserPoolException: Auth UserPool not configured."
+
+export default function ComponentX() {
+  return (
+    <div className="box">
+      ...
+    </div>
+  );
+}
+```
+
+This can also happen when using Next.js Layouts and calling Amplify APIs in child components at the module-level (at the top of your file/module). See below for an example of this issue:
+
+```javascript title="layout.tsx"
+import ConfigureAmplifyClientSide from '@/ConfigureAmplifyClientSide';
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <body className="container py-6">
+        <>
+          <ConfigureAmplifyClientSide />
+          {children}
+        </>
+      </body>
+    </html>
+  );
+}
+```
+
+```javascript title="ConfigureAmplifyClientSide.tsx"
+import { Amplify } from "aws-amplify";
+
+Amplify.configure(config, { ssr: true });
+
+export default function ConfigureAmplifyClientSide() {
+  return null;
+}
+```
+
+```javascript title="page.tsx"
+import { fetchAuthSession } from "aws-amplify/auth";
+
+// The layout calls configure, but fetchAuthSession ends up executing first
+// Will throw "AuthUserPoolException: Auth UserPool not configured."
+fetchAuthSession().then((session) => {
+  console.log(session);
+});
+
+export default function HomePage() {
+  return (
+    <div className="box">
+      ...
+    </div>
+  );
+}
+```
+
+## Configure Amplify on each page of a multi-page app
+
+If you are working in a multi-page app, make sure `Amplify.configure` is called for each page.
```

Comment:
What do you think of this alternative or similar:

If you are working in a multi-page app, you can call `Amplify.configure` in a common source file and then import it into each page.

Reply:
Ooo that is really nice